### PR TITLE
fix bug in Vector mode method

### PR DESCRIPTION
**Pull request by sunshineyyy**
This change will fix this issue. #65 
Also added Rspec test code to test the new Vector mode method.

### CODE REVIEW

File: spec/math/statistics/vector_spec.rb [hunk 1]
```diff
@@ -92,6 +92,9 @@
       context "#mode" do
         it "returns the mode" do
           @dv.mode
+          expect(@dv.mode).to eq(666)
+          mode_test_example = Daru::Vector.new [1,2,3,2,4,4,4,4], dtype: dtype
+          expect(mode_test_example.mode).to eq(4)
         end
```
Comment (v0dro) [inline spec/math/statistics/vector_spec.rb]: Thank you very much for this fix. Just one concern:

Could you please rewrite this whole test as:
``` ruby
mode_test_example = Daru::Vector.new [1,2,3,2,4,4,4,4], dtype: dtype
expect(mode_test_example.mode).to eq(4)
```

The above 2 lines aren't necessary anymore.

Comment (v0dro) [inline spec/math/statistics/vector_spec.rb]: Also don't forget to rebase your commits.

**Comment by sunshineyyy** [comment]
Sure. I can totally rewrite this. I am not super familiar with the rebase. Should I make another commit, then squash the two commits and submit another pull request?

**Comment by v0dro** [comment]
Just make another commit, squash them both into a single commit, and then push to your upstream with the `--force` flag. That should reflect it here.

**Comment by sunshineyyy** [comment]
I think it worked. Thanks!

**Pull request merged**